.github/workflows/api-docs.yml

```diff
@@ -0,0 +1,42 @@
+name: update-api-documentation
+
+on:
+  push:
+    branches: [master]
+    tags: ['v*']
+    paths:
+    - 'docs/**'
+
+jobs:
+  push-to-documentation-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Add key
```

Comment:
Is there a reason, why we are not using the `GITHUB_TOKEN` here? If I am not mistaken it has the correct permission.
https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#using-the-github_token-in-a-workflow

You can find an example of how we use it to create releases here: https://github.com/SAP/cloud-sdk/blob/eb89e31e5ab42a77f8a00b997cfa404d504f46c1/.github/workflows/build.yml#L70

Reply:
I don't exactly recall every small detail, but it seems a `GITHUB_TOKEN` enables auth only for GitHub actions that are provided by GitHub. For CLI inside of a container it doesn't work. For that reason, you need either an ssh key or a token from a developer who has merge rights to the repo. To me, a key seems like a more universal and independent way for repositories used by teams instead of individuals.

I have nothing against updating this pipeline by someone more experienced with the GH actions API and container configuration. I haven't found a good manual on this topic or definitive best practices. For example, I believe checkout action can be probably used instead of my clone CLI command, but I don't know if it makes a shallow checkout or not and what directory structure gonna be to sync everything. Also, for push, we'll have to provide our push token or key to a 3rd party, which might be a security concern.

Reply:
Unless it is a verified action I guess. But, ok, got it. And yes, we can also improve later on.